File: ООП/Homework2/animals_PY/classes/wilds/wolf.py
from .wild import Wild

class wolf(Wild):
    def __init__(self, name, height, weight, coloreye, place, date, leader):
        self.leader = leader
        super(wolf, self).__init__(name, height, weight, coloreye, place, date)

    def voice(self):
       print(f"{super().voice()} аууу")

    def __str__(self):
        if (self.leader == True):
            return f"{super().__str__()}\nСреда обитания: {self.place}\nДата нахождения: {self.date}\nВожак стаи"
        else:
            return f"{super().__str__()}\nСреда обитания: {self.place}\nДата нахождения: {self.date}"

File: ООП/Homework5/Зоопарк/all_animals/wilds/wolf.java
package Зоопарк.all_animals.wilds;

public class wolf extends wild{
    protected boolean leader;

    public wolf(double height, double weight, String eyeColor, String date, String place, boolean leader) {
        super(height, weight, eyeColor, date, place);
        this.leader = leader;
    }
    protected wolf(double height, double weight, String eyeColor, String date, String place) {
        super(height, weight, eyeColor, date, place);
        this.leader = false;
    }
    public boolean isLeader() {
        return this.leader;
    }
    public String sound() {
        return "Ауууу!";        
    }
    public String toString() {
        return String.format("Wolf (%s)", super.toString());
    }
}
